provisioning/device/src/client.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import * as Provisioning from './interfaces';
+import { ClientStateMachine } from './client_state_machine';
+import * as dbg from 'debug';
+const debug = dbg('azure-device-provisioning:device-client');
+
+export class ProvisioningDeviceClient implements Provisioning.ClientConfiguration {
+
+  /**
+   * User-Agent string passed to the service as part of communication
+   */
+  userAgent: string = 'azure-iot-node-provisioning'; // TODO
+
+  /**
+   * Default interval for polling, to use in case service doesn't provide it to us.
+   */
+  pollingInterval: number = 2000;
+
+  /**
+   * Default host for the provisioning service
+   */
+  provisioningHost: string = 'global.azure-devices-provisioning.net';
+
+  /**
+   * apiVersion to use while communicating with service.
+   */
+  apiVersion: string = '2017-08-31-preview';
+
+  /**
+   * default timeout to use when communicating with the service
+   */
+  timeoutInterval: number = 4000;
+
+  /**
+   * idScope to use when communicating with the provisioning service
+   */
+  idScope: string;
+
+  private _stateMachine: ClientStateMachine;
+
+  private constructor(transport: Provisioning.TransportHandlers, idScope: string) {
+    this.idScope = idScope;
+    transport.setClientConfig(this);
```

Comment:
>setClientConfig [](start = 14, length = 15)

hmm. so the transport constructor takes a config object and we let the user create the transport but then we overwrite it?

Reply:
yes, ugly.  refactoring.

---
In reply to: [148918155](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r148918155) [](ancestors = 148918155)